### PR TITLE
Webc 598 web checkout with just logging in

### PR DESCRIPTION
**Pull request by andre-kraus**

### CODE REVIEW

**Comment by augsteyer** [comment]
Also, **please cross-reference** if we need to remove the functionality of reading ```reserved_order_id``` as it should not show up anymore in responses.

File: src/app/code/community/Shopgate/Cloudapi/Helper/Frontend/Quote.php [hunk 1]
```diff
@@ -24,15 +24,20 @@ class Shopgate_Cloudapi_Helper_Frontend_Quote extends Mage_Core_Helper_Abstract
 {
     /**
      * @param Mage_Customer_Model_Customer $customer
+     * @param Mage_Core_Model_Store|null   $store
      *
      * @return Mage_Sales_Model_Quote
      * @throws Mage_Core_Model_Store_Exception
      */
-    public function createNewCustomerQuote($customer)
+    public function createNewCustomerQuote($customer, $store = null)
```
[the remark this answers is not in the document]
Comment (andre-kraus) [reply src/app/code/community/Shopgate/Cloudapi/Helper/Frontend/Quote.php]: Done

File: src/app/code/community/Shopgate/Cloudapi/Helper/Frontend/Quote.php [hunk 2]
```diff
@@ -24,15 +24,20 @@ class Shopgate_Cloudapi_Helper_Frontend_Quote extends Mage_Core_Helper_Abstract
 {
     /**
      * @param Mage_Customer_Model_Customer $customer
+     * @param Mage_Core_Model_Store|null   $store
      *
      * @return Mage_Sales_Model_Quote
      * @throws Mage_Core_Model_Store_Exception
```
[the remark this answers is not in the document]
Comment (andre-kraus) [reply src/app/code/community/Shopgate/Cloudapi/Helper/Frontend/Quote.php]: Sure

File: src/app/code/community/Shopgate/Cloudapi/Model/Api2/Carts/Utility.php [hunk 3]
```diff
@@ -142,6 +150,7 @@ private function loadInactiveCustomerQuote()
         $quote = Mage::getResourceModel('sales/quote_collection')
                      ->addFieldToSelect('*')
                      ->addFieldToFilter('customer_id', $this->getApiUser()->getUserId())
+                     ->addFieldToFilter('reserved_order_id', array('eq' => 'NULL'))
```
[the remark this answers is not in the document]
Comment (andre-kraus) [reply src/app/code/community/Shopgate/Cloudapi/Model/Api2/Carts/Utility.php]: Removed the method

File: src/app/code/community/Shopgate/Cloudapi/Helper/Frontend/Quote.php [hunk 4]
```diff
@@ -24,20 +24,17 @@ class Shopgate_Cloudapi_Helper_Frontend_Quote extends Mage_Core_Helper_Abstract
 {
     /**
      * @param Mage_Customer_Model_Customer $customer
-     * @param Mage_Core_Model_Store|null   $store
+     * @param Mage_Core_Model_Store        $store
      *
      * @return Mage_Sales_Model_Quote
      * @throws Mage_Core_Model_Store_Exception
```
Comment (augsteyer) [inline src/app/code/community/Shopgate/Cloudapi/Helper/Frontend/Quote.php]: this is now not thrown anymore

**Pull request merged**